### DISCUSSION TITLE
How to test component using useFieldArray

### DISCUSSION THREAD

**Discussion by marksmall** [react-hook-form/react-hook-form#2474] [Q&A]
I have a large form split into multiple components. One of the components is quite complex and uses dynamic fields, much like https://codesandbox.io/s/react-hook-form-usefieldarray-vy8fv from the docs.

My question is, how do I test my component using [@testing-library](https://testing-library.com/)?

I Render the component as you would normally i.e.

```javascript
const { getByText, queryByText, getAllByPlaceholderText } = render(
      <Items
        register={register}
        errors={errors}
        supplier={supplier}
        categories={categories}
      />,
      {
        wrapper: ({ children }) => (
          <FormProvider>{children}</FormProvider>
        ),
      },
    );
```
The problem is I get the error 

>TypeError: Cannot destructure property `isWatchAllRef` of 'undefined' or 'null'.

So my actual question is, How do I write my test, to take into account the `useFieldArray` hook? I've googled around to no avail, so any example you can point me to etc would be much appreciated, thanks.

Here is an actual screenshot of the error:

![useFieldArray-test](https://user-images.githubusercontent.com/1752272/89330753-a0ce6a80-d688-11ea-90c2-a7ec4fd9250d.png)

**Comment by bluebill1049** [comment]
Take a look our own repo, with useFieldArray.test you should get some good ideas there.

**Comment by lsbyerley** [comment]
@marksmall looking for some examples of this also. My component declares useFieldArray like this:

```
const { control } = useFormContext();
const { fields, append, remove } = useFieldArray({
    control,
    name,
  });
```

In my test, fields is empty. I'm not sure If i need to be mocking useFieldArray in my test or importing it directly? Lots of questions. Looking at the useFieldArray tests of this repo doesn't help too much

**Reply by marksmall** [reply]
I agree @lsbyerley I took a look this morning and none of what I came across help my situation. I've got a ridiculous time constraint on this work so have just disabled the tests for this component right now. Not great, but I gotta move on. I will re-visit this issue sometime soon, just not likely this week.

I don't want to **mock** `isFieldArray`, I think that is the wrong solution, I only mentioned it in my original post as I have no idea what to do. It feels wrong to me to have to pass in `isFieldArray` to child components, but that is all I've come across in the `react-hook-form` repository tests. I'm just having to leave this as an open problem for now.

**Reply by lsbyerley** [reply]
Understood, and yeah that doesn't seem right to me either. If i discover anything of note i'll be sure to pass it along
